src/components/Picklist/index.js

```diff
@@ -0,0 +1,401 @@
+        const { isOpen } = this.state;
+        const { className } = this.props;
+
+        return classnames(
+            'rainbow-picklist',
+            {
+                'rainbow-picklist--open': isOpen,
+            },
+            className,
+        );
+    }
+
+    getInputClassNames() {
+        const { icon } = this.getValue();
+        return classnames('rainbow-picklist_input', {
+            'rainbow-picklist_input--icon': !!icon,
+        });
+    }
+
+    getDropdownClassNames() {
+        const { isLoading } = this.props;
+        return classnames('rainbow-picklist_dropdown', {
+            'rainbow-picklist_dropdown--loading-box': isLoading,
+        });
+    }
+
+    getValue() {
+        const { value } = this.props;
+        if (value && typeof value === 'object') {
+            return value;
+        }
+        return {};
+    }
+
+    handleKeyUpPressed() {
+        const { activeOptionIndex } = this.state;
+        let nextActiveIndex;
+        if (activeOptionIndex < 1) {
+            nextActiveIndex = this.activeChildren.length - 1;
+        } else {
+            nextActiveIndex = (activeOptionIndex - 1) % this.activeChildren.length;
+        }
+        this.setState({
+            activeOptionIndex: nextActiveIndex,
+            activeOptionName: this.activeChildren[nextActiveIndex].name,
+        });
+    }
+
+    handleKeyDownPressed() {
+        const { activeOptionIndex } = this.state;
+        const nextActiveIndex = (activeOptionIndex + 1) % this.activeChildren.length;
+        this.setState({
+            activeOptionIndex: nextActiveIndex,
+            activeOptionName: this.activeChildren[nextActiveIndex].name,
+        });
+    }
+
+    handleKeyEnterPressed() {
+        const { onChange } = this.props;
+        const { activeOptionIndex } = this.state;
+        const { label, name, icon, value } = this.activeChildren[activeOptionIndex];
+        this.closeMenu();
+        return onChange({
+            label,
+            name,
+            icon,
+            value,
+        });
+    }
+
+    handleKeyPressed(event) {
+        const { isOpen } = this.state;
+        if (isOpen) {
+            if (event.keyCode === UP_KEY || event.keyCode === DOWN_KEY) event.preventDefault();
+            if (this.keyHandlerMap[event.keyCode]) {
+                return this.keyHandlerMap[event.keyCode]();
+            }
+        }
+        return null;
+    }
+
+    registerChild(childRef, childProps) {
+        const { childrenRefs } = this.state;
+        const [...nodes] = getChildMenuItemNodes(this.containerRef.current);
+        const newChildrenRefs = insertChildOrderly(childrenRefs, childRef, nodes);
+        const refIndex = newChildrenRefs.indexOf(childRef);
+        const extendedChildProps = {
+            ...childProps,
+            ref: childRef,
+        };
+        const { value } = this.props;
+        if (!isEmpty(value)) {
+            if (isEmpty(this.currentOption) && childProps.name === value.name) {
```

Comment:
I think you don't need to maintain this currentOption state since  it is much the same as value prop, instead make a helper function to compute what you need based on value prop, let's stick with functional model.